server/cromwell_routes.py

```diff
@@ -0,0 +1,185 @@
+import json
+import logging
+from datetime import datetime
+from shlex import quote
+from aiohttp import web
+
+import hailtop.batch as hb
+
+from util import (
+    get_analysis_runner_metadata,
+    get_email_from_request,
+    validate_output_dir,
+    check_dataset_and_group,
+    server_config,
+    publisher,
+    prepare_git_job,
+    run_batch_job_and_print_url,
+    cromwell_url,
+    PUBSUB_TOPIC,
+    DRIVER_IMAGE,
+)
+
+
+def add_cromwel_routes(
+    routes,
+):
+    def make_cromwell_url(suffix):
+        return cromwell_url + suffix
+
+    @routes.post('/cromwell')
+    async def cromwell(request):
+        email = get_email_from_request(request)
+        # When accessing a missing entry in the params dict, the resulting KeyError
+        # exception gets translated to a Bad Request error in the try block below.
+        params = await request.json()
+        try:
+            output_dir = validate_output_dir(params['output'])
+
+            dataset = params['dataset']
+            check_dataset_and_group(dataset, email)
+
+            repo = params['repo']
+            allowed_repos = server_config[dataset]['allowedRepos']
+            if repo not in allowed_repos:
+                raise web.HTTPForbidden(
+                    reason=(
+                        f'Repository "{repo}" is not one of the allowed repositories: '
+                        f'{", ".join(allowed_repos)}'
+                    )
+                )
+
+            access_level = params['accessLevel']
+            ds_config = server_config[dataset]
+            hail_token = ds_config.get(f'{access_level}Token')
+            project = ds_config.get(f'{access_level}Project')
+            service_account = ds_config.get(f'{access_level}ServiceAccount')
+            intermediate_dir = ds_config.get(f'{access_level}IntermediateDir')
```

Comment:
Wouldn't we just always put that in the temporary bucket?

Reply:
Is there one temporary bucket per dataset? Or one overall?